Libraries/PushNotificationIOS/RCTPushNotificationManager.m

```diff
@@ -305,4 +305,53 @@ - (void)handleRegisterUserNotificationSettings:(NSNotification *)notification
   resolve(RCTNullIfNil(initialNotification));
 }
 
+RCT_EXPORT_METHOD(getScheduledLocalNotifications:(RCTResponseSenderBlock)callback)
+{
+  NSArray<UILocalNotification *> *scheduledLocalNotifications = [UIApplication sharedApplication].scheduledLocalNotifications;
+
+
+  NSMutableArray *formattedScheduledLocalNotifications = [[NSMutableArray alloc] init];
+
+  for (UILocalNotification *notification in scheduledLocalNotifications) {
+
+    NSMutableDictionary *formattedScheduledLocalNotification = [NSMutableDictionary dictionary];
+
+    if (notification.fireDate) {
+      NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+      [formatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSSZZZZZ"];
+      NSString *fireDateString = [formatter stringFromDate:notification.fireDate];
+
+      formattedScheduledLocalNotification[@"fireDate"] = fireDateString;
+    }
+
+    if (notification.alertBody) {
```

Comment:
Pushed a refactor that moves the formatting loop to a class method and makes generous use of `RCTNullIfNil`.

Much cleaner now, thanks. 

Have a look and let me know if it's 👍